packages/trade-widget/src/App.tsx

```diff
@@ -0,0 +1,85 @@
+import { SignedOrder, ZeroEx } from '0x.js';
+import { InjectedWeb3Subprovider, RedundantRPCSubprovider } from '@0xproject/subproviders';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
+import {
+    Box,
+    Button,
+    Card,
+    CardContent,
+    CardHeaderTitle,
+    Column,
+    Columns,
+    Container,
+    Content,
+    Control,
+    Field,
+    Input,
+    Label,
+    TextArea,
+} from 'bloomer';
+import 'bulma/css/bulma.css';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import { applyMiddleware, compose, createStore, Store as ReduxStore } from 'redux';
+import thunk from 'redux-thunk';
+import * as Web3 from 'web3';
+import * as Web3ProviderEngine from 'web3-provider-engine';
+
+import './App.css';
+import { BuyWidget } from './containers/buy_widget';
+import './index.css';
+import registerServiceWorker from './registerServiceWorker';
+
+import { FixedProvider } from './liquidity_providers/fixed_provider';
+import { Dispatcher } from './redux/dispatcher';
+import { reducer, State } from './redux/reducer';
+import { BlockchainSaga } from './sagas/blockchain_saga';
+import { AssetToken } from './types';
+
+const TEST_NETWORK_ID = 50;
+const TEST_RPC = 'http://127.0.0.1:8545';
+
+const providerEngine = new Web3ProviderEngine();
+providerEngine.addProvider(new InjectedWeb3Subprovider((window as any).web3.currentProvider));
+providerEngine.addProvider(new RedundantRPCSubprovider([TEST_RPC]));
+providerEngine.start();
+
+const web3Wrapper = new Web3Wrapper(providerEngine);
+const zeroEx = new ZeroEx(providerEngine, { networkId: TEST_NETWORK_ID });
+
+const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const store: ReduxStore<State> = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));
```

Comment:
Really not a big fan of Thunks. Even Gaeron (creator) said it should only be used for small test projects... It introduces callback hell and doesn't handle well complex async requests.